features/device_key/source/DeviceKey.cpp

```diff
@@ -270,7 +270,10 @@ int DeviceKey::generate_root_of_trust()
 
     mbedtls_entropy_free(entropy);
     delete entropy;
-    ret = device_inject_root_of_trust(key_buff, actual_size);
+
+    if (ret == MBED_SUCCESS) {
```

Comment:
This is never going to happen. a few lines above `MBED_SUCCESS` is translated to `DEVICEKEY_SUCCESS` (or set to `DEVICEKEY_GENERATE_RANDOM_ERROR`). 
Or am I missing something?

Reply:
Thanks for that catch, fixed.